### PR TITLE
Improved error messages for vcat/hcat on BitArray

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by kshyatt** [comment]
Updated!!! @JeffBezanson can I get a review?

File: base/bitarray.jl [hunk 1]
```diff
@@ -1730,7 +1730,7 @@ function hcat(B::BitVector...)
     height = length(B[1])
     for j = 2:length(B)
         length(B[j]) == height ||
-            throw(DimensionMismatch("dimensions must match"))
+            throw(DimensionMismatch("dimensions must match: $j-th element has length $(length(B[j])), should have $height"))
```
Comment (JeffBezanson) [inline base/bitarray.jl]:
```suggestion
            throw(DimensionMismatch("dimensions must match: $j-th argument has length $(length(B[j])), should have $height"))
```
or "j-th array" or "j-th vector" etc.

**Comment by kshyatt** [comment]
Good to go?

**Comment by StefanKarpinski** [comment]
The concern is that capturing that info in the error might cause a slow-down in the method.

**Pull request merged**